File: hospitalManagement/Faculty.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hospitalManagement
{
    internal class Faculty
    {
        //Field
        private string id;
        // Infor field
        private string name;
        private string description;
        private string room;
        //  Management field
        private Employees employeeList;
        private Patients patientList;
        // Dynamic field
        // Properties

        public string Id { get => id; set => id = value; }
        public string Name { get => name; set => name = value; }
        public string Description { get => description; set => description = value; }
        public string Room { get => room; set => room = value; }
      
        internal Patients PatientList { get => patientList; set => patientList = value; }
        internal Employees EmployeeList { get => employeeList; set => employeeList = value; }

        // Constructors
        public Faculty()
        {
            employeeList = new Employees();
            patientList = new Patients();
           

        }
        public Faculty(string id, string name, string description, string room, Employees employeeList, Patients patientList)
        {
            this.id = id;
            this.name = name;
            this.description = description;
            this.room = room;
            this.employeeList = employeeList;
            this.patientList = patientList;
        }

        // Destructors
        ~Faculty() { }

        // Methods
        // in, output
        public virtual void Input()
        {
            Console.Write("ID: ");
            Id = Console.ReadLine();
            Console.Write("Name: ");
            Name = Console.ReadLine();
            Console.Write("Description: ");
            Description = Console.ReadLine();
        }


        public virtual void Output()
        {
            Console.WriteLine("The faculty");
            Console.WriteLine($"Id: {Id}");
            Console.WriteLine($"Name: {Name}");
            Console.WriteLine($"Description: ");
            Console.WriteLine($"Room: {Room}");
            Console.WriteLine($"The patients list: {patientList.Count}");
            patientList.ShowInformation();
            Console.WriteLine($"The employees list: {employeeList.Count}");
            employeeList.ShowInformation();
        }
        // General method
        // Other method
        // Overriding

        public void AddPatient()
        {

            this.PatientList.AddItem();
        }

        public void RemovePatient(string id)
        {
            this.PatientList.RemoveItem(id);
        }

        public void ShowAllPatients()
        {
            this.PatientList.ShowInformation();
        }

        public Patient FindPatient(string id)
        {
            return this.PatientList.FindItem(id);
        }

        public Patient UpdatePatient(string id)
        {
            return this.PatientList.UpdateItem(id);
        }

        public void AddEmployee()
        {
            this.EmployeeList.AddItem();
        }

        public void RemoveEmployee(string id)
        {
            this.EmployeeList.RemoveItem(id);
        }

        public void ShowAllEmployee()
        {
            this.EmployeeList.ShowInformation();
        }

        public Employee FindEmployee(string id)
        {
            return this.EmployeeList.FindItem(id);
        }

        public Employee UpdateEmployee(string id)
        {
            return this.EmployeeList.UpdateItem(id);
        }

        public void ClearPatient()
        {
            this.PatientList.Clear();
        }

        public void ClearEmployee()
        {
            this.EmployeeList.Clear();
        }
    }
}

File: src/hospitalManagement/Faculty.java
package hospitalManagement;

import java.util.Scanner;
public class Faculty
    {
        //Field
        private String id;
        // INFO field
        private String name;
        private String description;
        private String room;
        //  Management field
        private Doctors doctorList;
        private Patients patientList;
        // Dynamic field
        static Scanner input = new Scanner(System.in);
        // Properties(getter and setter)

        

        public String getId() {
            return id;
        }
        public void setId(String id) {
            this.id = id;
        }
        public String getName() {
            return name;
        }
        public void setName(String name) {
            this.name = name;
        }
        public String getDescription() {
            return description;
        }
        public void setDescription(String description) {
            this.description = description;
        }
        public String getRoom() {
            return room;
        }
        public void setRoom(String room) {
            this.room = room;
        }
        public Doctors getDoctorList() {
            return doctorList;
        }
        public void setDoctorList(Doctors doctorList) {
            this.doctorList = doctorList;
        }
        public Patients getPatientList() {
            return patientList;
        }
        public void setPatientList(Patients patientList) {
            this.patientList = patientList;
        }
        
        // Constructors
       
        public Faculty()
        {
            doctorList = new Doctors();
            patientList = new Patients();

        }
        public Faculty(String id, String name, String description, String room, Doctors doctorList,
                Patients patientList) {
            this.id = id;
            this.name = name;
            this.description = description;
            this.room = room;
            this.doctorList = doctorList;
            this.patientList = patientList;
        }

        // Destructors
 

        // Methods
        // in, output
        public void Input()
        {
            System.out.println("ID: ");
            id = System.in.toString();
            System.out.println("Name: ");
            name = System.in.toString();
            System.out.println("Description: ");
            description = System.in.toString();
        }

        public  void Input1()
        {
            System.out.println("ID: ");
            id = System.in.toString();
            System.out.println("Name: ");
            name = System.in.toString();
            System.out.println("Description: ");
            description = System.in.toString();
            System.out.println("List of doctors: ");
            System.out.println("Number of doctors: ");
            int i = input.nextInt();

            for (int j = 0; j < i; j++)
            {
                Doctor doctor = new Doctor();
                doctorList.AddItem(doctor);
            }

            
            System.out.println ("The patient list: ");
            System.out.println("Number of patient in charge: ");
            int k = input.nextInt();

            for (int j = 0; j < k; j++)
            {
                Patient patient = new Patient();
                patientList.AddItem(patient);
            }
        }

        public  void Output()
        {
            System.out.println("The faculty");
            System.out.format("Id: %s\n", this.id);
            System.out.format("Name: %s\n", this.name);
            System.out.format("Description: %s\n", this.description);
            System.out.format("Room: %s\n", this.room);
            System.out.format("The patients list: \n");
            patientList.ShowInformation();
            System.out.format("The doctors list: \n");
            doctorList.ShowInformation();
        }
        // General method
        // Other method
        // Overriding
    }